components/installer/pkg/config/installation-data.go

```diff
@@ -74,7 +77,18 @@ func NewInstallationData(installation *v1alpha1.Installation, installationConfig
 		EnableEtcdBackupOperator:   installationConfig.EnableEtcdBackupOperator,
 		EtcdBackupABSAccount:       installationConfig.EtcdBackupABSAccount,
 		EtcdBackupABSKey:           installationConfig.EtcdBackupABSKey,
+		Components:                 strings.Split(installationConfig.ComponentsList, ","),
 		IsLocalInstallation:        isLocalInstallationFunc,
 	}
 	return res, nil
 }
+
+func (installationData *InstallationData) ShouldInstallComponent(componentName string) bool {
```

Comment:
IMO it can be easily optimized

```go
func NewInstallationData(installation *v1alpha1.Installation, installationConfig *installationConfig) (*InstallationData, error) {

	isLocalInstallationFunc := func() bool {
		return installationConfig.ExternalIPAddress == ""
	}

	convertToMap := func(cList string) map[string]struct{} {
		splitted := strings.Split(cList, ",")
		out := make(map[string]struct{}, len(splitted))

		for _, c := range splitted {
			out[c] = struct{}{}
		}

		return out
	}

	ctx := InstallationContext{
		Name:      installation.Name,
		Namespace: installation.Namespace,
	}

	res := &InstallationData{
		// other fields...
		Components:                 convertToMap(installationConfig.ComponentsList),
		IsLocalInstallation:        isLocalInstallationFunc,
	}
	return res, nil
}

func (installationData *InstallationData) ShouldInstallComponent(componentName string) bool {
	_, found := installationData.Components[componentName]
	return found
}
```

thanks to that the `ShouldInstallComponent` always have O(1) complexity. Additionally all redundant items will be indexed only once.

Reply:
Done!